### PR TITLE
=str Fixup some docs in GraphStage.

### PR DESCRIPTION
**Pull request by He-Pin**

### CODE REVIEW

File: akka-stream/src/main/scala/akka/stream/stage/GraphStage.scala [hunk 1]
```diff
@@ -534,11 +534,9 @@ abstract class GraphStageLogic private[stream] (val inCount: Int, val outCount:
...
       case Attributes.CancellationStrategy.AfterDelay(delay, _) =>
         // since the port is not actually cancelled, we install a handler to ignore upcoming elements
-        connection.inHandler = new InHandler {
-          // ignore pushs now, since the stage wanted it cancelled already
-          override def onPush(): Unit = ()
-          // do not ignore termination signals
-        }
+        // ignore pushs now, since the stage wanted it cancelled already
+        // do not ignore termination signals
+        connection.inHandler = EagerTerminateInput
```
Comment (hepin1989) [inline akka-stream/src/main/scala/akka/stream/stage/GraphStage.scala:539]: reduce an allocation

File: akka-stream/src/main/scala/akka/stream/stage/GraphStage.scala [hunk 2]
```diff
@@ -667,8 +665,8 @@ abstract class GraphStageLogic private[stream] (val inCount: Int, val outCount:
    */
   final protected def complete[T](out: Outlet[T]): Unit =
     getHandler(out) match {
-      case e: Emitting[_] => e.addFollowUp(new EmittingCompletion(e.out, e.previous))
-      case _              => interpreter.complete(conn(out))
+      case e: Emitting[T @unchecked] => e.addFollowUp(new EmittingCompletion[T](e.out, e.previous))
+      case _                         => interpreter.complete(conn(out))
     }
```
Comment (hepin1989) [inline akka-stream/src/main/scala/akka/stream/stage/GraphStage.scala:670]: IDEA report red here

File: akka-stream/src/main/scala/akka/stream/stage/GraphStage.scala [hunk 3]
```diff
@@ -383,7 +383,7 @@ abstract class GraphStageLogic private[stream] (val inCount: Int, val outCount:
    * The [[akka.stream.Materializer]] that has set this GraphStage in motion.
    *
    * Can not be used from a `GraphStage` constructor. Access to materializer is provided by the
-   * [[akka.stream.scaladsl.Source.setup]], [[akka.stream.scaladsl.Flow.setup]] and [[akka.stream.scaladsl.Sink.setup]]
+   * [[akka.stream.scaladsl.Source.fromMaterializer]], [[akka.stream.scaladsl.Flow.fromMaterializer]] and [[akka.stream.scaladsl.Sink.fromMaterializer]]
```
Comment (hepin1989) [inline akka-stream/src/main/scala/akka/stream/stage/GraphStage.scala:386]: replace the deprecated method

**Review by johanandren — APPROVED** [review]

**Comment by hepin1989** [comment]
@raboof ping~

**Review by raboof — APPROVED** [review]

**Pull request merged**